cluster/terraform_kubernetes/kube_state_metrics.tf

```diff
@@ -0,0 +1,219 @@
+
+
+resource "kubernetes_service_account" "kube_state_metrics" {
+  metadata {
+    name      = "kube-state-metrics"
+    namespace = "monitoring"
+    labels = {
+      "app.kubernetes.io/component" = "exporter"
+      "app.kubernetes.io/name"      = "kube-state-metrics"
+      "app.kubernetes.io/version"   = var.kube_state_metrics_version
+    }
+  }
+}
+
+
+resource "kubernetes_cluster_role" "kube_state_metrics" {
+  metadata {
+    name = "kube-state-metrics"
+    labels = {
+      "app.kubernetes.io/component" = "exporter"
+      "app.kubernetes.io/name"      = "kube-state-metrics"
+      "app.kubernetes.io/version"   = var.kube_state_metrics_version
+    }
+  }
+
+  rule {
+    api_groups = [""]
+    resources  = ["configmaps", "secrets", "nodes", "pods", "services", "resourcequotas", "replicationcontrollers", "limitranges", "persistentvolumeclaims", "persistentvolumes", "namespaces", "endpoints"]
+    verbs      = ["get", "list", "watch"]
+  }
+
+  rule {
+    api_groups = ["certificates.k8s.io"]
+    resources  = ["certificatesigningrequests"]
+    verbs      = ["list", "get", "watch"]
+  }
+  rule {
+    api_groups = ["batch"]
+    resources  = ["jobs"]
+    verbs      = ["get", "list", "watch"]
+  }
+  rule {
+    api_groups = ["apps"]
+    resources  = ["deployments", "replicasets", "statefulsets", "daemonsets"]
+    verbs      = ["list", "watch"]
+  }
+  rule {
+    api_groups = ["storage.k8s.io"]
+    resources  = ["volumeattachments"]
+    verbs      = ["get", "list", "watch"]
+  }
+  rule {
+    api_groups = ["admissionregistration.k8s.io"]
+    resources  = ["mutatingwebhookconfigurations", "validatingwebhookconfigurations"]
+    verbs      = ["get", "list", "watch"]
+  }
+  rule {
+    api_groups = ["coordination.k8s.io"]
+    resources  = ["leases"]
+    verbs      = ["get", "list", "watch"]
+  }
+  rule {
+    api_groups = ["autoscaling"]
+    resources  = ["horizontalpodautoscalers"]
+    verbs      = ["get", "list", "watch"]
+  }
+  rule {
+    api_groups = ["policy"]
+    resources  = ["poddisruptionbudgets"]
+    verbs      = ["get", "list", "watch"]
+  }
+  rule {
+    api_groups = ["batch"]
+    resources  = ["cronjobs"]
+    verbs      = ["get", "list", "watch"]
+  }
+  rule {
+    api_groups = ["", "apps", "batch", "networking.k8s.io", "policy", "autoscaling", "certificates.k8s.io", "coordination.k8s.io", "storage.k8s.io"]
```

Comment:
if we have added the api_groups separately above, can we remove the duplicates from here?

Reply:
the separate ones removed. That is the only way it worked but also looked tidier anyways